packages/application-generic/src/services/filters/filters.service.ts

```diff
@@ -0,0 +1,152 @@
+import * as _ from 'lodash';
+import { StepFilter, SubscriberEntity, TenantEntity } from '@novu/dal';
+import {
+  FieldOperatorEnum,
+  FILTER_TO_LABEL,
+  IBaseFieldFilterPart,
+  ICondition,
+  ITenantDefine,
+  ITriggerPayload,
+} from '@novu/shared';
+
+export interface IFilterVariables {
+  payload?: ITriggerPayload;
+  subscriber?: SubscriberEntity;
+  webhook?: Record<string, unknown>;
+  tenant?: TenantEntity;
+}
+
+export class FilterProcessingDetails {
+  private conditions: ICondition[] = [];
+  private filter: StepFilter;
+  private variables: IFilterVariables;
+
+  addFilter(filter: StepFilter, variables: IFilterVariables) {
+    this.filter = filter;
+    this.variables = variables;
+    this.conditions = [];
+  }
+
+  addCondition(condition: ICondition) {
+    this.conditions.push(condition);
+  }
+
+  toObject() {
+    return {
+      payload: this.variables,
+      filter: this.filter,
+      conditions: this.conditions,
+    };
+  }
+
+  toString() {
+    return JSON.stringify(this.toObject());
+  }
+}
+
+export class FilterService {
+  public operate(
+    actualValue: any,
+    operator: FieldOperatorEnum,
+    filterValue: any
+  ): boolean {
+    if (operator === FieldOperatorEnum.EQUAL) {
+      return actualValue === filterValue;
+    }
+    if (operator === FieldOperatorEnum.NOT_EQUAL) {
+      return actualValue !== filterValue;
+    }
+    if (operator === FieldOperatorEnum.LARGER) {
+      return actualValue > filterValue;
+    }
+    if (operator === FieldOperatorEnum.SMALLER) {
+      return actualValue < filterValue;
+    }
+    if (operator === FieldOperatorEnum.LARGER_EQUAL) {
+      return actualValue >= filterValue;
+    }
+    if (operator === FieldOperatorEnum.SMALLER_EQUAL) {
+      return actualValue <= filterValue;
+    }
+    if (operator === FieldOperatorEnum.NOT_IN) {
+      return actualValue.includes(filterValue);
+    }
+    if (operator === FieldOperatorEnum.IN) {
+      return actualValue.includes(filterValue);
+    }
+    if (operator === FieldOperatorEnum.IS_DEFINED) {
+      return actualValue !== undefined;
+    }
+
+    return false;
+  }
```

Comment:
Isolate this in a function will help a lot when implementing the payload filtering for the Broadcast.